### PR TITLE
Create test for Firewalld

### PR DESCRIPTION
**Pull request by atanno**
Create a test for basic FirewallD functionality

Created the first version for this test.

- Related ticket: https://progress.opensuse.org/issues/47159
- Verification run: http://10.100.51.204/tests/76#

### CODE REVIEW

File: tests/console/firewalld.pm [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+
+    # Test #1 - Stop firewalld then start it
+    record_info 'Test #1', 'Test: Stop firewalld, then start it';
+    systemctl('stop firewalld');
+    systemctl('start firewalld');
+
+    # Check Service State, enable it if necessary
+    record_info 'Check Service State';
+    assert_script_run("if [ `systemctl is-active firewalld | grep -q -i 'inactive' ; echo $?` -eq 0 ]; then systemctl start firewalld; fi");
```
Comment (dzedro) [inline tests/console/firewalld.pm]: `if systemctl is-active firewalld | grep -q -i 'inactive'; then systemctl start firewalld; fi`
can be shorter and do the same

File: tests/console/firewalld.pm [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    assert_script_run("firewall-cmd --zone=public --add-protocol=icmp");
+    assert_script_run("iptables -C IN_public_allow -p icmp -m conntrack --ctstate NEW -j ACCEPT");
+
+    assert_script_run("firewall-cmd --zone=public --add-port=2000-3000/udp");
+    assert_script_run("iptables -C IN_public_allow -p udp --dport 2000:3000 -m conntrack --ctstate NEW -j ACCEPT");
+
+    # Flush rules
+    record_info 'Flush Rules';
+    assert_script_run("firewall-cmd --reload");
```
Comment (dzedro) [inline tests/console/firewalld.pm]: Why should --reload flush the firewall rules ?

File: tests/console/firewalld.pm (hunk 1 above)
Comment (atanno) [reply tests/console/firewalld.pm]: I do agree with you. I'm checking this test's code and changing it according to your suggestion.

File: tests/console/firewalld.pm (hunk 2 above)
Comment (atanno) [reply tests/console/firewalld.pm]: As these rules are temporary, --reload would "flush" them. But reading your comment, I have to agree that it looks weird and it should read "Reload default configuration".

**Comment by dzedro** [comment]
Please squash the commits

**Pull request closed without merging**